readme.md

```diff
@@ -399,7 +400,7 @@ use whatever casing style is best and transform the output to camelCase.
 
 There are also two very general patterns to notice:
 
-- No values are used as keys.
+- No values are used as keys (unless a nested property has an object as the value).
```

Comment:
Let's talk about this in person. At this point don't want to add this and I know it's happening in practice. :wink: I think we can use some HAL-style resource embedding to work around this nicely.